internal/pkg/cli/svc_init.go

```diff
@@ -604,24 +621,45 @@ func (o *initSvcOpts) askSvcPort() (err error) {
 		return nil
 	}
 
-	port, err := o.prompt.Get(
-		fmt.Sprintf(svcInitSvcPortPrompt, color.Emphasize("port")),
+	if o.wkldType == manifestinfo.RequestDrivenWebServiceType {
+		selectedPorts, err := o.prompt.Get(
+			fmt.Sprintf(svcInitSvcPortPrompt, color.Emphasize("port")),
+			svcInitSvcPortHelpPrompt,
+			validateSvcPort,
+			prompt.WithDefaultInput(defaultPort),
+			prompt.WithFinalMessage("Port:"),
+		)
+		if err != nil {
+			return fmt.Errorf("get port: %w", err)
+		}
+		if len(strings.Split(selectedPorts, ",")) > 1 {
+			return fmt.Errorf("App Runner Service doesn't expose multiple ports")
+		}
+		o.ports = make([]string, 1)
+		o.ports[0] = selectedPorts
+		return nil
+	}
+
+	selectedPorts, err := o.prompt.Get(
+		fmt.Sprintf(svcInitSvcPortPrompt, color.Emphasize("port(s)")),
 		svcInitSvcPortHelpPrompt,
 		validateSvcPort,
 		prompt.WithDefaultInput(defaultPort),
-		prompt.WithFinalMessage("Port:"),
+		prompt.WithFinalMessage("Port(s):"),
 	)
 	if err != nil {
 		return fmt.Errorf("get port: %w", err)
 	}
+	portList := strings.Split(selectedPorts, ",")
+	o.ports = make([]string, len(portList))
+	copy(o.ports, portList)
+	return nil
+}
 
-	portUint, err := strconv.ParseUint(port, 10, 16)
-	if err != nil {
-		return fmt.Errorf("parse port string: %w", err)
+func (o *initSvcOpts) validateMultiplePorts() error {
+	if (o.wkldType != manifestinfo.LoadBalancedWebServiceType && o.wkldType != manifestinfo.BackendServiceType) && len(o.ports) > 1 {
+		return fmt.Errorf("%s doesn't expose multiple ports", o.wkldType)
```

Comment:
If this error occurs when we attempt to specify multiple ports on services that aren't LBWS or BS, can we change the wording?
```suggestion
		return fmt.Errorf("%s cannot expose multiple ports", o.wkldType)
```